### PR TITLE
Add comments and rename method name to improve code readability

### PR DESCRIPTION
**Pull request by Kourin1996**
# Description

This PR adds some comments, fixes the codes without any impact, and renames some method names to improve code readability

## Explanation about some changes

Keeps a record of the reason of some changes.

### Message code used in committed seal

> Committed seal is calculated by each of the validator signing the hash along with COMMIT_MSG_CODE message code of its private key

`COMMIT_MSG_CODE` is 1 according to EIP-650, but it's 2 in Quorum implementation because they added PREPREPARE_MSG_CODE. PolygonSDK is using same set of message codes to Quorum implementation. So we can calculate committed seal by 2.

https://docs.goquorum.consensys.net/en/stable/Concepts/Consensus/IBFT/

### Remaining gas check in apply of Transition

This PR removes following code:
https://github.com/0xPolygon/polygon-sdk/blob/2f50a194f21d972b542ccd1e1571ccba40c5c810/state/executor.go#L373-L376

The reason is that `leftoverGas` is calculated by `msg.gas - (constant value) - (size of Input with non-zero value)`, which means leftoverGas is always less than msg.gas. So will never enter the branch.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [x] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: consensus/ibft/sign.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func ecrecoverFromHeader(h *types.Header) (types.Address, error) {
 }
 
 func signSealImpl(prv *ecdsa.PrivateKey, h *types.Header, committed bool) ([]byte, error) {
-	sig, err := signHash(h)
+	sig, err := getHeaderHash(h)
```
Comment (Vuksan) [inline consensus/ibft/sign.go]: Please change variable sig to a more proper name, since it's not a signature:
```suggestion
	hash, err := getHeaderHash(h)
```

File: consensus/ibft/sign.go [hunk 2]
```diff
@@ -110,8 +110,7 @@ func writeCommittedSeals(h *types.Header, seals [][]byte) (*types.Header, error)
 	return h, nil
 }
 
-// TODO: This function is called signHash, we are doing no signing
-func signHash(h *types.Header) ([]byte, error) {
+func getHeaderHash(h *types.Header) ([]byte, error) {
```
Comment (Vuksan) [inline consensus/ibft/sign.go]: What do you think if we rename this to `calculateHeaderHash()`?
As it is now, I would expect it to only do `return h.Hash`

File: state/txn.go [hunk 3]
```diff
@@ -333,7 +334,7 @@ func (txn *Txn) GetState(addr types.Address, key types.Hash) types.Hash {
 		}
 	}
 
-	// Under which condition is this called?
+	// Get account state from trie tree
```
Comment (Vuksan) [inline state/txn.go]: Can you please elaborate a bit here (in code comment) on what's the difference between `trie tree` and `radix tree` up on the line 327?

File: state/executor.go [hunk 4]
```diff
@@ -370,10 +365,6 @@ func (t *Transition) apply(msg *types.Transaction) (
 	if err != nil {
 		return nil, 0, false, err
 	}
-	// TODO: Check if this is even possible
-	if leftoverGas > msg.Gas {
-		return nil, 0, false, errorVMOutOfGas
-	}
```
Comment (Vuksan) [inline state/executor.go]: I'm afraid we cannot do this, it is the only check that we have if intrinsic gas cost goes above `msg.Gas`.
If you take a look at `preCheck()` method on line 349 it deducts the `msg.Gas` from `intrinsicGas` cost of the transaction it calculated.

If the intrinsic gas is greater than the gas limit that sender has provided through `msg.Gas`, then `gasAvailable` will go through underflow, and end up being some big number.
That's why we still need this weird check on the outside to check if the `leftoverGas` is greater than `msg.Gas` because then we know that what has occurred is that transaction has run out of gas during intrinsic gas cost check.

Please leave this as it is, I have a fix for this as a part of my task 🙂

**Review by Vuksan — CHANGES_REQUESTED** [review]
Mostly minor issues

File: state/executor.go (hunk 4 above)
Comment (Vuksan) [reply state/executor.go]: What saddens me is that our tests still pass, even with this issue. 🙃

File: consensus/ibft/sign.go (hunk 2 above)
Comment (lazartravica) [reply consensus/ibft/sign.go]: Agree with Vuksan on this one, `get` should only be reserved for getter methods and simple helper funcs.

File: state/txn.go (hunk 3 above)
Comment (lazartravica) [reply state/txn.go]: Also, try to include a comment like: `If the object was not found in the radix trie (due to reasons XXX), we fetch it from the trie tre...`

In simpler terms, we should try to comment what is the logic behind these if branches.

**Review by lazartravica — CHANGES_REQUESTED** [review]

**Review by lazartravica — APPROVED** [review]
💯

**Review by Vuksan — APPROVED** [review]
👍

**Review by zivkovicmilos — APPROVED** [review]
Thank you for clarifying some tricky parts of the code, it is much more understandable now what is happening 💯
